ccmlib/scylla_node.py

```diff
@@ -549,8 +549,15 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
 
         MB = 1024 * 1024
 
-        def process_opts(opts, args_translation_map = {}):
+        def process_opts(opts, args_translation_map = {}, allowed_duplicates_set = set()):
             ext_args = OrderedDict()
+            def should_add_option(key):
```

Comment:
the nested function is more readable, but it also renders this method more complicated, how about this?

```py
if key.startswith("--scylla-manager"):
    continue

if key in allowed_duplicates_set:
    ext_args.get(key, []).append(val)
elif key not in ext_args:
    ext_args[key] = val
```

for following reasons:

1. the combinations of conditions is getting convoluted in this function, better off simplifying them
2. `dict.set_default()` helps to consolidate the logic, but it hurts the readability.
3. we should not use list for all values of options. the boxing-unboxing hurts the readability.